<launch>
    <arg name="yolo_model_path" default="$(find tj2_yolo)/models/crescendo_2024-03-22.torchscript" />
    <arg name="yolo_class_names_path" default="$(find tj2_yolo)/models/crescendo_2024.names" />


    <group ns="tj2">
        <node name="tj2_yolo" pkg="tj2_yolo" type="tj2_yolo" output="screen" required="false">
            <param name="model_path" value="$(arg yolo_model_path)" />
            <param name="classes_path" value="$(arg yolo_class_names_path)" />

            <param name="confidence_threshold" value="0.6" />
            <param name="nms_iou_threshold" value="0.4" />

            <param name="warmup_image_width" value="1280" />
            <param name="warmup_image_height" value="720" />

            <param name="publish_overlay" value="true" />
            <param name="report_loop_times" value="true" />
            <param name="min_depth" value="0.1" />
            <param name="max_depth" value="5.0" />
            <param name="erosion_size" value="11" />
            <param name="depth_num_std_devs" value="2.5" />
            <param name="relative_threshold" value="50" />
            <param name="long_loop_warning_ms" value="100" />
            <param name="message_delay_warning_ms" value="500" />
            <param name="visuals_cube_opacity" value="0.75" />
            <param name="visuals_line_width" value="0.01" />
            <param name="use_depth" value="false" />

            <remap from="color/image_raw" to="/tj2_zed/rgb/image_rect_color" />
            <remap from="depth/image_raw" to="/tj2_zed/depth/depth_registered" />
            <remap from="color/camera_info" to="/tj2_zed/rgb/camera_info" />
        </node>
    </group>
    <include file="$(find tj2_yolo)/launch/label_publisher.launch">
        <arg name="class_names_path" default="$(arg yolo_class_names_path)" />
    </include>
    <node name="publish_video" pkg="tj2_yolo" type="publish_video.py" output="screen"
        required="false">
        <param name="path"
            value="/home/tj2/tj2_ros/dataset_builder/data/Crescendo 2024 Durham College - GoPro 3 [4Jiatb8Pr-o].mp4" />
        <remap from="image" to="/tj2_zed/rgb/image_rect_color" />
    </node>
</launch>